src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -170,14 +172,41 @@ public ConfidentialClientApplicationBuilder WithClientAssertion(string signedCli
         /// This is a delegate that computes a Base-64 encoded JWT for each authentication call.</param>
         /// <returns>The ConfidentialClientApplicationBuilder to chain more .With methods</returns>
         /// <remarks> Callers can use this mechanism to cache their assertions </remarks>
+        [Obsolete("Use WithClientAssertion(Func<CancellationToken, Task<string>> clientAssertionAsyncDelegate)")]
```

Comment:
Do we need to obsolete this one? Why not keep both overrides?

Reply:
Ok